### PR TITLE
General: Replace GC_ALIGN macros with alignas

### PR DESCRIPTION
**Pull request by lioncash**
Standard-supported alignment facilities ∴ out with compiler-specific.

### CODE REVIEW

File: Source/Core/Core/PowerPC/PowerPC.h [hunk 1]
```diff
@@ -57,7 +57,7 @@ struct tlb_entry
 };
 
 // This contains the entire state of the emulated PowerPC "Gekko" CPU.
-struct GC_ALIGNED64(PowerPCState)
+struct alignas(16) PowerPCState
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/PowerPC.h]: Important to note that this is changed to 16, as the only used instance of it is aligned to 16 bytes (and it's forbidden by the standard to use a smaller alignment for a type that has a stricter alignment).

Comment (phire) [inline Source/Core/Core/PowerPC/PowerPC.h]: Probably someone wanted a stricter alignment to force this structure to start on a 64 byte cacheline. 

The only actual alignment requirement is the paired single registers must be 16 byte aligned.

Comment (lioncash) [inline Source/Core/Core/PowerPC/PowerPC.h]: So the align on the struct itself can be removed then?

Comment (phire) [inline Source/Core/Core/PowerPC/PowerPC.h]: yes???

**Comment by lioncash** [comment]
Is there anything else holding this back?

**Comment by phire** [comment]
No, everything looks good to me.

**Pull request merged**